cmd/releasego/get-asset-version.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+
+	"github.com/microsoft/go-infra/buildmodel"
+	"github.com/microsoft/go-infra/buildmodel/buildassets"
+	"github.com/microsoft/go-infra/goversion"
+	"github.com/microsoft/go-infra/subcmd"
+)
+
+func init() {
+	subcommands = append(subcommands, new(getAssetVersionCmd))
+}
+
+type getAssetVersionCmd struct{}
+
+func (g getAssetVersionCmd) Name() string {
+	return "get-asset-version"
+}
+
+func (g getAssetVersionCmd) Summary() string {
+	return "Gets the asset version out of a build asset JSON file."
+}
+
+func (g getAssetVersionCmd) Description() string {
+	return `
+
+Log the asset version found in the given build asset JSON file. Optionally print an AzDO logging
+command to set a variable that later release automation steps can use.
+
+This command can also validate that the build asset JSON version matches an expected string version.
+Exits 0 if the given version number matches the content of the build asset JSON file, or 1 if it
+doesn't. This is used in release automation as a safety check.
+`
+}
+
+func (g getAssetVersionCmd) Handle(p subcmd.ParseFunc) error {
+	buildAssetJSON := flag.String("build-asset-json", "", "[Required] The path of a build asset JSON file to read.")
+
+	validateVersionFlag := flag.String(
+		"version", "",
+		"A Microsoft-built Go version, in 1.2.3-1[-fips] format.\n"+
+			"If specified, must match the build asset JSON's version or this command will fail.\n"+
+			"The string 'nil' is treated as the same as not setting the value, for use in CI where empty string can't be used.\n"+
+			"Optionally use this to validate expectations.")
+
+	setVariable := flag.String("set", "", "An AzDO variable name to set.")
```

Comment:
nit: can we call this setVariable or something that makes it more obvious what it does from logs when you don't have the help?

Reply:
Sure, going one step further and changing it to `-set-azdo-variable`. (It's not like length is really a concern here, since it's pretty useless outside of CI. 😄)